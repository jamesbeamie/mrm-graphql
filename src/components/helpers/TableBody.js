import React from 'react';
import PropTypes from 'prop-types';
import { formatRoomData } from '../../graphql/mappers/Rooms';
import Room from '../rooms/Room';

/**
 * Component that builds the body of a table
 *
 * @param {array} props
 *
 * @returns {JSX}
 */
const TableBody = props => (
  <div className="table_body">
    {props.rooms &&
      props.rooms.map(room => (
        <Room
          currentPage={props.currentPage}
          room={formatRoomData(room)}
          key={room.id}
          locations={props.location}
          refetch={props.refetch}
        />
      ))}
  </div>
);
TableBody.propTypes = {
  rooms: PropTypes.arrayOf(PropTypes.object).isRequired,
  location: PropTypes.arrayOf(PropTypes.object).isRequired,
  currentPage: PropTypes.number,
  refetch: PropTypes.func,
};

TableBody.defaultProps = {
  currentPage: 1,
  refetch: null,
};

export default TableBody;
