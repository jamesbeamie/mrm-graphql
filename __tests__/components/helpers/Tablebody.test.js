import React from 'react';
import { shallow } from 'enzyme';
import TableBody from '../../../src/components/helpers/TableBody';

describe('Pagination Component', () => {
  const props = {
    rooms: [
      {
        id: '3',
        name: 'Gulu',
        floor: {
          block: {
            offices: {
              name: 'The Crest',
              location: {
                name: 'Kampala',
                id: '1',
              },
            },
          },
        },
      },
    ],
    location: [],
  };
  const props2 = {
    rooms: [
      {
        id: '3',
        name: 'Gulu',
        floor: {
          block: { },
        },
      },
    ],
    location: [],
  };

  const bodyWrapper = shallow(<TableBody {...props} />);
  const bodyWrapper2 = shallow(<TableBody {...props2} />);

  it('renders without crashing', () => {
    expect(bodyWrapper).toMatchSnapshot();
  });
  it('renders without crashing', () => {
    expect(bodyWrapper2).toMatchSnapshot();
  });
});

