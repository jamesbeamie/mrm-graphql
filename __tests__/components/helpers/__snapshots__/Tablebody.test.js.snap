// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination Component renders without crashing 1`] = `
<div
  className="table_body"
>
  <Room
    currentPage={1}
    key="3"
    locations={Array []}
    refetch={null}
    room={
      Object {
        "id": "3",
        "location": "Kampala",
        "locationId": "1",
        "name": "Gulu",
        "office": "The Crest",
      }
    }
  />
</div>
`;

exports[`Pagination Component renders without crashing 2`] = `
<div
  className="table_body"
>
  <Room
    currentPage={1}
    key="3"
    locations={Array []}
    refetch={null}
    room={
      Object {
        "id": "3",
        "location": "",
        "locationId": "",
        "name": "Gulu",
        "office": "",
      }
    }
  />
</div>
`;
